aequilibrae/project/network/osm_builder.py

```diff
@@ -46,37 +52,28 @@ def doWork(self):
             conn = sqlite3.connect(self.path)
             self.conn = spatialite_connection(conn)
         self.curr = self.conn.cursor()
-
+        self.__worksetup()
         node_count = self.data_structures()
-        nodes_to_add, node_ids = self.importing_links(node_count)
-        self.import_nodes(nodes_to_add, node_ids)
+        self.importing_links(node_count)
+        self.__emit_all(["finished_threaded_procedure", 0])
 
     def data_structures(self):
-
-        osmi = []
-        logger.info("Consolidating geo elements")
-        self.__emit_all(["text", "Consolidating geo elements"])
-        self.__emit_all(["maxValue", len(self.osm_items)])
-
-        for i, x in enumerate(self.osm_items):
-            osmi.append(x["elements"])
-            self.__emit_all(["Value", i])
-        self.osm_items = sum(osmi, [])
-
         logger.info("Separating nodes and links")
         self.__emit_all(["text", "Separating nodes and links"])
         self.__emit_all(["maxValue", len(self.osm_items)])
 
         alinks = []
         n = []
-        for i, x in enumerate(self.osm_items):
-            if x["type"] == "way":
-                alinks.append(x)
-            elif x["type"] == "node":
-                n.append(x)
-            self.__emit_all(["Value", i])
+        tot_items = len(self.osm_items)
+        for i in range(tot_items, 0, -1):
+            item = self.osm_items.pop(-1)
+            if item['type'] == "way":
+                alinks.append(item)
+            elif item['type'] == "node":
+                n.append(item)
+            self.__emit_all(["Value", tot_items - i])
+        gc.collect()
```

Comment:
This is modifying osm_items as it loops through them?
Are they memory hungry?
I'd suggest putting some comments around this loop